src/lib/y2storage/devicegraph.rb

```diff
@@ -361,5 +389,35 @@ def device_tree
       writer = Y2Storage::YamlWriter.new
       writer.yaml_device_tree(self)
     end
+
+    # Removes a device, all its descendants and also the potential orphans of
+    # all the removed devices.
+    #
+    # @see Device#potential_orphans
+    #
+    # @param device [Device]
+    # @param keep [Array<Device>] used to control the recursive calls
+    #
+    # @raise [ArgumentError] if the device does not exist in the devicegraph
+    def remove_with_dependants(device, keep: [])
+      raise(ArgumentError, "No device provided") if device.nil?
+      raise(ArgumentError, "Device not found") unless device.exists_in_devicegraph?(self)
+
+      children = device.children
+      until children.empty?
+        remove_with_dependants(children.first, keep: keep + [device])
+        children = device.children
+      end
+
+      orphans = device.respond_to?(:potential_orphans) ? device.potential_orphans : []
```

Comment:
is it needed? all devices should have this method, so why to ask for it? and above you check it is not nil.

Reply:
It's maybe a little bit too defensive, but just imagine something changes in libstorage-ng and, for whatever reason we get a non-wrapped object. I.e. we get a `Storage::Whatever` because we have still not created the corresponding `Y2Storage::Whatever` and `Storage::Whatever` is not a subclass of `Storage::Device`. It may sounds paranoid, but better be safe from libstorage-ng changes.